.github/workflows/tests.yml

```
@@ -112,6 +112,17 @@ jobs:
       fail-fast: false
     steps:
       - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          cache: pip
+          cache-dependency-path: requirements-tests.txt
+      - run: |
+          pip install tomli pathspec packaging
```

As @JelleZijlstra says, it might be easier just to use `requirements-tests.txt` here. If the `pip install` step of our CI starts getting slow then we can rethink this approach, but at the moment I think it's fast enough, and it's less maintenance work to just rely on a single `requirements.txt` file. That way we don't need to update this file if we add a new third-party dependency to `tests/utils.py`; we'll only need to update `requirements-tests.txt`.

```suggestion
          pip install requirements-tests.txt
```

---

Installing a bunch of unrelated and unnecessary dependencies was indeed my concern. I was also curious what's the actual bare-minimum.

Come to worse we can also split it in another requirements file that is referenced by `requirements-tests.txt`.
Both of both world with minimal maintenance and installs.

---

> We could also split it in another requirements file that is referenced by `requirements-tests.txt`.
> Both of both world with minimal maintenance and installs.

Yup, I do this in `typeshed-stats`: https://github.com/AlexWaygood/typeshed-stats/blob/main/requirements/all.txt

It still feels like it's adding complexity in terms of cognitive load, though, and I'm still not _convinced_ it's necessary. The `pip install` step takes up quite a high percentage of the total CI time for typeshed-stats, so it was worth micro-optimising it! Here, I'm not so sure.